optimum/onnx/auto/configuration_onnx_auto.py

```diff
@@ -0,0 +1,211 @@
+    CONFIG_MAPPING,
+    MODEL_NAMES_MAPPING,
+    AutoConfig,
+    _get_class_name,
+    _LazyConfigMapping,
+)
+from transformers.models.auto.dynamic import get_class_from_dynamic_module
+from transformers.onnx.config import OnnxConfig
+from transformers.utils import logging
+
+
+logger = logging.get_logger(__name__)
+
+ONNX_CONFIG_MAPPING_NAMES = OrderedDict(
+    [
+        # Add configs here
+        ("albert", "AlbertOnnxConfig"),
+        ("bart", "BartOnnxConfig"),
+        ("bert", "BertOnnxConfig"),
+        ("camembert", "CamembertOnnxConfig"),
+        ("distilbert", "DistilBertOnnxConfig"),
+        ("gpt2", "GPT2OnnxConfig"),
+        ("ibert", "IBertOnnxConfig"),
+        ("longformer", "LongformerOnnxConfig"),
+        ("mbart", "MBartOnnxConfig"),
+        ("roberta", "RobertaOnnxConfig"),
+        ("t5", "T5OnnxConfig"),
+        ("xlm-roberta", "XLMRobertaOnnxConfig"),
+        ("xlm-roberta-xl", "XLMRobertaXLOnnxConfig"),
+    ]
+)
+
+
+def config_class_to_model_type(config):
+    """Converts a onnx config class name to the corresponding model type"""
+    for key, cls in ONNX_CONFIG_MAPPING_NAMES.items():
+        if cls == config:
+            return key
+    return None
+
+
+ONNX_CONFIG_MAPPING = _LazyConfigMapping(ONNX_CONFIG_MAPPING_NAMES)
+
+
+def _list_model_options(indent, config_to_class=None, use_model_types=True):
+    if config_to_class is None and not use_model_types:
+        raise ValueError("Using `use_model_types=False` requires a `config_to_class` dictionary.")
+    if use_model_types:
+        if config_to_class is None:
+            model_type_to_name = {
+                model_type: f"[`{config}`]" for model_type, config in ONNX_CONFIG_MAPPING_NAMES.items()
+            }
+        else:
+            model_type_to_name = {
+                model_type: _get_class_name(model_class)
+                for model_type, model_class in config_to_class.items()
+                if model_type in MODEL_NAMES_MAPPING
+            }
+        lines = [
+            f"{indent}- **{model_type}** -- {model_type_to_name[model_type]} ({MODEL_NAMES_MAPPING[model_type]} model)"
+            for model_type in sorted(model_type_to_name.keys())
+        ]
+    else:
+        config_to_name = {
+            ONNX_CONFIG_MAPPING_NAMES[config]: _get_class_name(clas)
+            for config, clas in config_to_class.items()
+            if config in ONNX_CONFIG_MAPPING_NAMES
+        }
+        config_to_model_name = {
+            config: MODEL_NAMES_MAPPING[model_type] for model_type, config in ONNX_CONFIG_MAPPING_NAMES.items()
+        }
+        lines = [
+            f"{indent}- [`{config_name}`] configuration class: {config_to_name[config_name]} ({config_to_model_name[config_name]} model)"
+            for config_name in sorted(config_to_name.keys())
+        ]
+    return "\n".join(lines)
+
+
+def replace_list_option_in_docstrings(config_to_class=None, use_model_types=True):
```

Comment:
The purpose of this decorator is to add the list of supported models in the docstring?

Reply:
Yes, I forgot to add back the docstring, will do!